### ISSUE TITLE
[tool_crash] _TypeError: (#0      _ProgramInfoBuilder.createInfoNodeFor (package:vm_snapshot_analysis/v8_profile.dart:419:75))

### ISSUE THREAD

**Issue by joranmulderij** [flutter/flutter#132695]
## Command
```
flutter build apk --analyze-size --target-platform android-arm
```

## Steps to Reproduce

Not really sure. Medium size project, and just wanted to know how much my app weighed 😄 

Same thing happened when building for windows with analyze-size.

## Logs
_TypeError: (#0      _ProgramInfoBuilder.createInfoNodeFor (package:vm_snapshot_analysis/v8_profile.dart:419:75))
```
#0      _ProgramInfoBuilder.createInfoNodeFor (package:vm_snapshot_analysis/v8_profile.dart:419:75)
#1      _ProgramInfoBuilder.getInfoNodeFor (package:vm_snapshot_analysis/v8_profile.dart:374:14)
#2      ListIterable.forEach (dart:_internal/iterable.dart:39:13)
#3      _ProgramInfoBuilder.build (package:vm_snapshot_analysis/v8_profile.dart:338:16)
#4      toProgramInfo (package:vm_snapshot_analysis/v8_profile.dart:282:8)
#5      _treemapFromSnapshot (package:vm_snapshot_analysis/treemap.dart:158:27)
#6      treemapFromJson (package:vm_snapshot_analysis/treemap.dart:85:5)
#7      SizeAnalyzer.analyzeZipSizeAndAotSnapshot (package:flutter_tools/src/base/analyze_size.dart:123:59)
#8      AndroidGradleBuilder._performCodeSizeAnalysis (package:flutter_tools/src/android/gradle.dart:567:60)
#9      AndroidGradleBuilder.buildGradleApp (package:flutter_tools/src/android/gradle.dart:548:15)
<asynchronous suspension>
#10     AndroidGradleBuilder.buildApk (package:flutter_tools/src/android/gradle.dart:208:5)
<asynchronous suspension>
#11     BuildApkCommand.runCommand (package:flutter_tools/src/commands/build_apk.dart:116:5)
<asynchronous suspension>
#12     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1336:27)
<asynchronous suspension>
#13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#14     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#15     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:296:9)
<asynchronous suspension>
#16     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
```
```
[32m[✓][39m Flutter (Channel stable, 3.10.6, on Microsoft Windows [Version 10.0.22621.2134], locale en-US)
    [32m•[39m Flutter version 3.10.6 on channel stable at C:\Users\joran\flutter\flutter
    [32m•[39m Upstream repository https://github.com/flutter/flutter.git
    [32m•[39m Framework revision f468f3366c (5 weeks ago), 2023-07-12 15:19:05 -0700
    [32m•[39m Engine revision cdbeda788a
    [32m•[39m Dart version 3.0.6
    [32m•[39m DevTools version 2.23.1

[32m[✓][39m Windows Version (Installed version of Windows is version 10 or higher)

[32m[✓][39m Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    [32m•[39m Android SDK at C:\Users\joran\AppData\Local\Android\sdk
    [32m•[39m Platform android-33, build-tools 33.0.0
    [32m•[39m Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    [32m•[39m Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    [32m•[39m All Android licenses accepted.

[32m[✓][39m Chrome - develop for the web
    [32m•[39m Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[32m[✓][39m Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.6)
    [32m•[39m Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    [32m•[39m Visual Studio Community 2022 version 17.2.32630.192
    [32m•[39m Windows 10 SDK version 10.0.19041.0

[32m[✓][39m Android Studio (version 2021.2)
    [32m•[39m Android Studio at C:\Program Files\Android\Android Studio
    [32m•[39m Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    [32m•[39m Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    [32m•[39m Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[32m[✓][39m VS Code (version 1.80.1)
    [32m•[39m VS Code at C:\Users\joran\AppData\Local\Programs\Microsoft VS Code
    [32m•[39m Flutter extension version 3.70.0

[32m[✓][39m Connected device (3 available)
    [32m•[39m Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.2134]
    [32m•[39m Chrome (web)      • chrome  • web-javascript • Google Chrome 115.0.5790.171
    [32m•[39m Edge (web)        • edge    • web-javascript • Microsoft Edge 115.0.1901.203
    [33m![39m Device emulator-5562 is offline.

[32m[✓][39m Network resources
    [32m•[39m All expected network resources are available.

[32m•[39m No issues found!

```

## Flutter Application Metadata
**Type**: app
**Version**: 1.0.0+1
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: stable
**Creation framework version**: f468f3366c26a5092eb964a230ce7892fda8f2f8
### Plugins
audioplayers-4.1.0
audioplayers_android-3.0.2
audioplayers_darwin-4.1.0
audioplayers_linux-2.1.0
audioplayers_web-3.1.0
audioplayers_windows-2.0.2
desktop_multi_window-0.2.0
dynamic_color-1.6.6
file_picker-5.3.0
flutter_inappwebview-5.7.2+3
flutter_native_splash-2.3.0
flutter_plugin_android_lifecycle-2.0.15
fullscreen_window-1.0.4
integration_test
path_provider-2.0.15
path_provider_android-2.0.27
path_provider_foundation-2.2.3
path_provider_linux-2.1.10
path_provider_windows-2.1.6
share_plus-7.0.2
shared_preferences-2.2.0
shared_preferences_android-2.1.4
shared_preferences_foundation-2.2.2
shared_preferences_linux-2.2.0
shared_preferences_web-2.1.0
shared_preferences_windows-2.2.0
sqflite-2.2.8+4
sqlite3_flutter_libs-0.5.15
url_launcher-6.1.12
url_launcher_android-6.0.37
url_launcher_ios-6.1.4
url_launcher_linux-3.0.5
url_launcher_macos-3.0.6
url_launcher_web-2.0.18
url_launcher_windows-3.0.7

**Comment by mkustermann** [comment]
Seems similar to https://github.com/flutter/flutter/issues/129706

@joranmulderij  Could you try on newest stable channel (3.13.0)?

**Comment by joranmulderij** [comment]
I upgraded Flutter, ran flutter clean, and built for windows with --analyze-size.

<details><summary>Crash report</summary>
<p>

Flutter crash report.
Please report a bug at https://github.com/flutter/flutter/issues.

## command

flutter build windows --analyze-size

## exception

_TypeError: Null check operator used on a null value

```
#0      _ProgramInfoBuilder.createInfoNodeFor (package:vm_snapshot_analysis/v8_profile.dart:419:75)
#1      _ProgramInfoBuilder.getInfoNodeFor (package:vm_snapshot_analysis/v8_profile.dart:374:14)
#2      ListIterable.forEach (dart:_internal/iterable.dart:39:13)
#3      _ProgramInfoBuilder.build (package:vm_snapshot_analysis/v8_profile.dart:338:16)
#4      toProgramInfo (package:vm_snapshot_analysis/v8_profile.dart:282:8)
#5      _treemapFromSnapshot (package:vm_snapshot_analysis/treemap.dart:158:27)
#6      treemapFromJson (package:vm_snapshot_analysis/treemap.dart:85:5)
#7      SizeAnalyzer.analyzeAotSnapshot (package:flutter_tools/src/base/analyze_size.dart:61:59)
#8      buildWindows (package:flutter_tools/src/windows/build_windows.dart:123:60)
<asynchronous suspension>
#9      BuildWindowsCommand.runCommand (package:flutter_tools/src/commands/build_windows.dart:56:5)
<asynchronous suspension>
#10     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1297:27)
<asynchronous suspension>
#11     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#12     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#13     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:339:9)
<asynchronous suspension>
#14     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#15     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:285:5)
<asynchronous suspension>
#16     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:115:9)
<asynchronous suspension>
#17     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#18     main (package:flutter_tools/executable.dart:90:3)
<asynchronous suspension>
```

## flutter doctor

```
[32m[✓][39m Flutter (Channel stable, 3.13.0, on Microsoft Windows [Version 10.0.22621.2134],
    locale en-US)
    [32m•[39m Flutter version 3.13.0 on channel stable at C:\Users\joran\flutter\flutter
    [32m•[39m Upstream repository https://github.com/flutter/flutter.git
    [32m•[39m Framework revision efbf63d9c6 (2 days ago), 2023-08-15 21:05:06 -0500
    [32m•[39m Engine revision 1ac611c64e
    [32m•[39m Dart version 3.1.0
    [32m•[39m DevTools version 2.25.0

[32m[✓][39m Windows Version (Installed version of Windows is version 10 or higher)

[32m[✓][39m Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    [32m•[39m Android SDK at C:\Users\joran\AppData\Local\Android\sdk
    [32m•[39m Platform android-33, build-tools 33.0.0
    [32m•[39m Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    [32m•[39m Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    [32m•[39m All Android licenses accepted.

[32m[✓][39m Chrome - develop for the web
    [32m•[39m Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[32m[✓][39m Visual Studio - develop Windows apps (Visual Studio Community 2022 17.2.6)
    [32m•[39m Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    [32m•[39m Visual Studio Community 2022 version 17.2.32630.192
    [32m•[39m Windows 10 SDK version 10.0.19041.0

[32m[✓][39m Android Studio (version 2021.2)
    [32m•[39m Android Studio at C:\Program Files\Android\Android Studio
    [32m•[39m Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    [32m•[39m Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    [32m•[39m Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[32m[✓][39m VS Code (version 1.81.1)
    [32m•[39m VS Code at C:\Users\joran\AppData\Local\Programs\Microsoft VS Code
    [32m•[39m Flutter extension version 3.70.0

[32m[✓][39m Connected device (3 available)
    [32m•[39m Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version
      10.0.22621.2134]
    [32m•[39m Chrome (web)      • chrome  • web-javascript • Google Chrome 115.0.5790.171
    [32m•[39m Edge (web)        • edge    • web-javascript • Microsoft Edge 115.0.1901.203
    [33m![39m Device emulator-5562 is offline.

[32m[✓][39m Network resources
    [32m•[39m All expected network resources are available.

[32m•[39m No issues found!
```


</p>
</details>

**Comment by sstrickl** [comment]
So looking at this, the source location given for the error suggests that you're running a version of `vm_snapshot_analysis` prior to `0.7.4`... and in fact, if I check the [https://github.com/flutter/flutter/tree/stable](stable) branch, the version of `vm_snapshot_analysis` in the `pubspec.yaml` for `flutter_tools` is `0.7.2`. Given that `pkg/flutter_tools/pubspec.yaml` was updated to depend on `vm_snapshot_analysis: 0.7.6` in https://github.com/flutter/flutter/pull/130163 on 7. July, I'm surprised that update didn't end up in 3.13.0.

While I look into that, I've checked the 3.14.0 candidates and it _is_ updated there.  If you're comfortable trying out the beta channel and seeing whether that fixes your issue, then I'd appreciate it to verify that there's no issues when using the newest `vm_snapshot_analysis` package, but meanwhile I'll see what we can do about the stable channel.
